### PR TITLE
Add Verify() to verify the database structure

### PR DESCRIPTION
**Pull request by oschwald**

### CODE REVIEW

File: traverse.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		reader: r,
+		nodes: []netNode{
+			netNode{
+				ip: make(net.IP, s),
+			},
+		},
+	}
+}
+
+// Next prepared the next network for reading with the Network method. It
```
Comment (autarch) [inline traverse.go]: s/prepared/prepares/

**Comment by autarch** [comment]
I think this only works because of the particular way the MMDB writer is implemented. There's no reason that the data section couldn't start with a chunk of data that isn't referenced by the search tree, but _is_ referenced as a pointer later in the data section.

That doesn't happen right now because of the fact that the writer deduplicates data as it serialized data that is associated with a search tree node.

That said, it's probably fine to merge this, we just need to remember to change this code if we change the writer.

**Comment by oschwald** [comment]
@autarch, this is what I was talking about yesterday. Currently it checks that each offset is in the search tree. The Perl version worked in the same way. Finding unused data sections generally would be much more difficult.

**Comment by oschwald** [comment]
Merged by hand.

**Pull request closed without merging**